test/LibraryManager.IntegrationTest/LibmanCompletionTests.cs

```diff
@@ -33,5 +32,152 @@ public void LibmanCompletion_Destination()
+
+            Editor.KeyboardCommands.Type("\"library\":");
+            LibmanTestsUtility.WaitForCompletionEntries(Editor, expectedCompletionEntries, caseInsensitive: true, timeout: 5000);
+
+            Editor.KeyboardCommands.Type("j");
+            LibmanTestsUtility.WaitForCompletionEntryNotPresent(Editor, "bootstrap", caseInsensitive: true);
+
+            Editor.KeyboardCommands.Type("q");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "jquery", caseInsensitive: true);
+        }
+
+        [TestMethod]
+        [Ignore("Ignored for bug 628945")]
+        public void LibmanCompletion_LibraryVersionForCsnjs()
+        {
+            _libManConfig.Open();
+
+            Editor.Caret.MoveToBeginningOfFile();
+            Editor.Caret.MoveDown(4);
+            Editor.KeyboardCommands.Type("\"provider\": \"cdnjs\",");
+            Editor.KeyboardCommands.Enter();
+
+            Editor.KeyboardCommands.Type("\"library\":");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "jquery", caseInsensitive: true, timeout: 5000);
+
+            Editor.KeyboardCommands.Type("jquery@");
+            LibmanTestsUtility.WaitForCompletionEntries(Editor, new string[] { }, caseInsensitive: true);
+        }
+
+        [TestMethod]
+        public void LibmanCompletion_LibraryForFilesystem()
+        {
+            _libManConfig.Open();
+            string[] expectedCompletionEntries = new[] {
+                "Properties/",
+                "wwwroot/",
+            };
+
+            Editor.Caret.MoveToBeginningOfFile();
+            Editor.Caret.MoveDown(4);
+            Editor.KeyboardCommands.Type("\"provider\": \"filesystem\",");
+            Editor.KeyboardCommands.Enter();
+
+            Editor.KeyboardCommands.Type("\"library\":");
+            LibmanTestsUtility.WaitForCompletionEntries(Editor, expectedCompletionEntries, caseInsensitive: true);
+
+            Editor.KeyboardCommands.Type("w");
+            LibmanTestsUtility.WaitForCompletionEntryNotPresent(Editor, "Properties/", caseInsensitive: true);
+
+            Editor.KeyboardCommands.Type("w");
+            LibmanTestsUtility.WaitForCompletionEntry(Editor, "wwwroot/", caseInsensitive: true);
+        }
+
+        [TestMethod]
+        [Ignore("Ignored for bug 629065")]
+        public void LibmanCompletion_LibraryForUnpkg()
+        {
+            _libManConfig.Open();
+            string[] expectedCompletionEntries = new[] {
+                "bootstrap",
+                "jquery",
+            };
+
+            Editor.Caret.MoveToBeginningOfFile();
+            Editor.Caret.MoveDown(4);
+            Editor.KeyboardCommands.Type("\"provider\": \"unpkg\",");
+            Editor.KeyboardCommands.Enter();
+
+            Editor.KeyboardCommands.Type("\"library\":");
+            LibmanTestsUtility.WaitForCompletionEntries(Editor, expectedCompletionEntries, caseInsensitive: true, timeout: 5000);
+
+            Editor.KeyboardCommands.Type("j");
```

Comment:
The rest of this test is just completion filtering.

Reply:
will remove

---
In reply to: [193926027](https://github.com/aspnet/LibraryManager/pull/108#discussion_r193926027) [](ancestors = 193926027)